x-pack/plugins/observability_solution/inventory/public/components/search_bar/index.tsx

```diff
@@ -51,11 +53,47 @@ export function SearchBar() {
     syncSearchBarWithUrl();
   }, [syncSearchBarWithUrl]);
 
+  const registerSearchSubmittedEvent = useCallback(
+    ({
+      searchQuery,
+      searchIsUpdate,
+      searchEntityTypes,
+    }: {
+      searchQuery: Query | undefined;
+      searchEntityTypes: EntityType[] | undefined;
+      searchIsUpdate: boolean | undefined;
+    }) => {
+      telemetry.reportEntityInventorySearchQuerySubmitted({
+        kuery_fields: getKqlFieldNamesFromExpression(searchQuery?.query as string),
```

Comment:
what happens if I pass an invalid kuery ?

Reply:
Thank you for pointing this out. I didn't check the internal behavior of `getKqlFieldNames`, so I didn't notice that there could be error cases. 

I will add a try/catch and pass the attribute as an empty array in case of an error.

Reply:
https://github.com/elastic/kibana/pull/197055/commits/0851b1ca457c8f22cc6e5c56b1b3e8a25424e219